reference-manual/src/main/asciidoc/pipeline-api/pipeline-api.adoc

```diff
@@ -422,6 +422,43 @@ include::{javasource}/BuildComputation.java[tag=s16a]
 In a similar fashion you can integrate other external systems with a
 Jet pipeline.
 
+== Machine Learning Model Prediction
+Usage of pre-trained Machine Learning models are common to see in the
+pipelines as an external scoring/classification mechanisms. 
+Hazelcast Jet comes with the facility that enables uploading files or 
+directories along with the job for later retrieval in the 
+service objects. The transformation step can take advantage of having a 
+local service instance with leveraging the uploaded resources to build the 
+model serving service. This eliminates the management cost of an
+external model serving service. Below is an example of using pre-trained
+http://docs.h2o.ai/h2o/latest-stable/h2o-docs/productionizing.html#what-is-a-mojo[H2O MOJO file] 
+for prediction on Hazelcast Jet pipeline:  
+
+[source]
+----
+include::{javasource}/MLModelPrediction.java[tag=s1]
+----
+
+After configuring the job to include the MOJO file, it can be accessed
+in the context object passed in to the service factory via the {jet-javadoc}/core/ProcessorSupplier.Context.html#attachedFile-java.lang.String-[`attachedFile()`]
+or {jet-javadoc}/core/ProcessorSupplier.Context.html#attachedDirectory-java.lang.String-[`attachedDirectory()`]
+method. That method returns the file or directory locally which has been 
```

Comment:
We should mention that this file/directory is stored in `/tmp` locally. 

Should not we also mention some information about attaching big files? Even if handling big files can be from user PoV only technical detail (like that they are split into `128k` chunks and store into maps) users still need some information about how many memory is needed for cluster which will be used for it (i.e. - without implementation knowledge it is not intuitive that with default Jet configuration where each map has 1 backup you actually need 3 nodes with `1GB` memory to load `1GB` file; I think that it can be confusing why just not only 2 nodes with `1GB` memory are not enough).

Reply:
May be that detail should go into to the Javadoc of the attachFile/Directory methods ? WDYT ?

Reply:
Yes, I think they can be described in Javadoc of those methods.